### PR TITLE
Hardening 3

### PR DESCRIPTION
**Pull request by gregaubert**
* [x] Fix bottom borders in rules page facets
* [x] Fix wrong treemap legend for coverage measure
* [x] Fix bad state of facets' arrows on measures page
* [x] [SONAR-9605](https://jira.sonarsource.com/browse/SONAR-9605)
* [x] [SONAR-9434](https://jira.sonarsource.com/browse/SONAR-9434)
* [x] [SONAR-9390](https://jira.sonarsource.com/browse/SONAR-9390)
* [x] [SONAR-9435](https://jira.sonarsource.com/browse/SONAR-9435)

### CODE REVIEW

File: server/sonar-web/src/main/js/components/SourceViewer/SourceViewerBase.js [hunk 1]
```diff
@@ -470,6 +475,19 @@ export default class SourceViewerBase extends React.PureComponent {
     }
   };
 
+  handlePopupToggle = (issue /*: string */, popupName /*: string */, open /*: ?boolean */) => {
+    this.setState((state /*: State */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/SourceViewer/SourceViewerBase.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/apps/issues/components/App.js [hunk 2]
```diff
@@ -572,6 +599,19 @@ export default class App extends React.PureComponent {
     });
   };
 
+  handlePopupToggle = (issue /*: string */, popupName /*: string */, open /*: ?boolean */) => {
+    this.setState((state /*: State */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/issues/components/App.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-web/src/main/js/apps/quality-gates/components/Details.js [hunk 3]
```diff
@@ -80,9 +81,15 @@ export default class Details extends React.PureComponent {
     const { qualityGate, onSetAsDefault, onUnsetAsDefault } = this.props;
 
     if (qualityGate.isDefault) {
-      unsetQualityGateAsDefault(qualityGate.id).then(() => onUnsetAsDefault(qualityGate));
+      unsetQualityGateAsDefault(qualityGate.id).then(
+        () => onUnsetAsDefault(qualityGate),
+        throwGlobalError
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/quality-gates/components/Details.js]: _This is just a notice for future._ We need to be consistent with this `throwGlobalError`: I add it in api, you add it in components. I'm fine with both solutions, but we need to have only one way.

File: server/sonar-web/src/main/js/apps/coding-rules/templates/facets/_coding-rules-facet-header.hbs [hunk 4]
```diff
@@ -1,4 +1,6 @@
-<a class="search-navigator-facet-header js-facet-toggle">
-  <i class="icon-checkbox {{#if enabled}}icon-checkbox-checked{{/if}}"></i>
-  {{t 'coding_rules.facet' property}}
-</a>
+<span class="search-navigator-facet-header">
+  <a class="js-facet-toggle">
+    <i class="icon-checkbox {{#if enabled}}icon-checkbox-checked{{/if}}"></i>
+    {{t 'coding_rules.facet' property}}
+  </a>
+</span>
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/coding-rules/templates/facets/_coding-rules-facet-header.hbs]: Ah yeah good catch, I updated my configuration of vscode! Thanks!

File: server/sonar-web/src/main/js/apps/quality-gates/components/Details.js (hunk 3 above)
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/quality-gates/components/Details.js]: Ah my bad, I think you are right, it should always be in the api.

**Pull request merged**